src/server/pps/server/api_server.go

```diff
@@ -2658,7 +2658,19 @@ func (a *apiServer) inspectPipeline(ctx context.Context, pipeline *pps.Pipeline,
 					return nil, errors.EnsureStack(err)
 				}
 			} else {
-				info.Details.Service.Ip = service.Spec.ClusterIP
+				if info.Details.Service.Type == "LoadBalancer" {
+					//GCP: service.Status.LoadBalancer.Ingress[0].IP
+					//AWS: service.Status.LoadBalancer.Ingress[0].Hostname
+					if len(service.Status.LoadBalancer.Ingress) == 1 {
```

Comment:
I think that you expect `len(service.Status.LoadBalancer.Ingress)` to be either zero or one here; is it worth explicitly erroring if it is two or more?

Reply:
We chatted about this offline. While having two more more in an exceptional condition, it's not an error condition. ie. we don't want to blow up the `inspectPipeline` call if there's too many load balancer IPs. I'm also not aware of any kube implementation that returns more than one.